src/HIE/Bios/Cradle.hs

```diff
@@ -390,45 +390,53 @@ type GhcProc = (FilePath, [String])
 -- generate a fake GHC that can be passed to cabal
 -- when run with --interactive, it will print out its
 -- command-line arguments and exit
-getCabalWrapperTool :: GhcProc -> FilePath -> IO FilePath
-getCabalWrapperTool (ghcPath, ghcArgs) wdir = do
-  wrapper_fp <-
-    if isWindows
-      then do
-        cacheDir <- getCacheDir ""
-        let srcHash = show (fingerprintString cabalWrapperHs)
-        let wrapper_name = "wrapper-" ++ srcHash
-        let wrapper_fp = cacheDir </> wrapper_name <.> "exe"
-        exists <- doesFileExist wrapper_fp
-        unless exists $ withSystemTempDirectory "hie-bios" $ \ tmpDir -> do
-            createDirectoryIfMissing True cacheDir
-            let wrapper_hs = cacheDir </> wrapper_name <.> "hs"
-            writeFile wrapper_hs cabalWrapperHs
-            let ghc = (proc ghcPath $
-                        ghcArgs ++ ["-rtsopts=ignore", "-outputdir", tmpDir, "-o", wrapper_fp, wrapper_hs])
-                        { cwd = Just wdir }
-            readCreateProcess ghc "" >>= putStr
-        return wrapper_fp
-      else writeSystemTempFile "bios-wrapper" cabalWrapper
+withCabalWrapperTool :: GhcProc -> FilePath -> (FilePath -> IO a) -> IO a
+withCabalWrapperTool (ghcPath, ghcArgs) wdir k = do
+  if isWindows
+    then do
+      cacheDir <- getCacheDir ""
+      let srcHash = show (fingerprintString cabalWrapperHs)
+      let wrapper_name = "wrapper-" ++ srcHash
+      let wrapper_fp = cacheDir </> wrapper_name <.> "exe"
+      exists <- doesFileExist wrapper_fp
+      unless exists $ withSystemTempDirectory "hie-bios" $ \ tmpDir -> do
+          createDirectoryIfMissing True cacheDir
+          let wrapper_hs = cacheDir </> wrapper_name <.> "hs"
+          writeFile wrapper_hs cabalWrapperHs
+          let ghc = (proc ghcPath $
+                      ghcArgs ++ ["-rtsopts=ignore", "-outputdir", tmpDir, "-o", wrapper_fp, wrapper_hs])
+                      { cwd = Just wdir }
+          readCreateProcess ghc "" >>= putStr
+      setMode wrapper_fp
+      k wrapper_fp
+    else withSystemTempFile "bios-wrapper"
+            (\loc h -> do
+                        hPutStr h cabalWrapper
+                        hClose h
+                        setMode loc
+                        k loc)
+
+  where
+    setMode wrapper_fp = do
+      setFileMode wrapper_fp accessModes
+      _check <- readFile wrapper_fp
```

Comment:
Should we just delete that?

Reply:
I think we could, I had a look at where it was introduced and it used to print out the cradle to stderr after writing it to the file. That was modified when we started using a proper executable on windows to get the arguments.